### PR TITLE
Adds LINQ challenges

### PR DESCRIPTION
**Pull request by MathewBeldon**
Adds LINQ challenges

Broken down into three sections

Easy
Intermediate
Hard

### CODE REVIEW

File: Challenges/LINQ/LINQ.Challenge.Tests/HardTests.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
+﻿using FluentAssertions;
+using LINQ.Challenge.Models;
+using LINQ.Challenge.Models.DTOs;
+
+namespace LINQ.Challenge.Tests;
+
+public class HardTests
```
Comment (stephen-atkinson) [inline Challenges/LINQ/LINQ.Challenge.Tests/HardTests.cs:7]: Would be good to have some tests that require multiple linq statements
E.g. List of people aged 25 or over, ordered alphabetically, returned as a PersonDto - Where().OrderBy().Select().ToArray()

Or a add up all the distinct numbers that are greater than 50
Where().Distinct().Sum()

File: Challenges/LINQ/LINQ.Challenge.Tests/IntermediateTests.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
+﻿using FluentAssertions;
+
+namespace LINQ.Challenge.Tests;
+
+public class IntermediateTests
```
Comment (stephen-atkinson) [inline Challenges/LINQ/LINQ.Challenge.Tests/IntermediateTests.cs:5]: Could we add a distinct here?

File: Challenges/LINQ/LINQ.Challenge.Tests/EasyTests.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
+using FluentAssertions;
+using LINQ.Challenge.Models;
+
+namespace LINQ.Challenge.Tests;
+
+public class EasyTests
```
[the remark this answers is not in the document]
Comment (MathewBeldon) [reply Challenges/LINQ/LINQ.Challenge.Tests/EasyTests.cs:6]: Added to easy - Any, Max and Min 
Added to Intermediate - Sum of all odd numbers

File: Challenges/LINQ/LINQ.Challenge.Tests/HardTests.cs (hunk 1 above)
Comment (MathewBeldon) [reply Challenges/LINQ/LINQ.Challenge.Tests/HardTests.cs:7]: Added to Hard - Unique and return as DTO, Get an order list of unique first names of people over a specified age

File: Challenges/LINQ/LINQ.Challenge.Tests/IntermediateTests.cs (hunk 2 above)
Comment (MathewBeldon) [reply Challenges/LINQ/LINQ.Challenge.Tests/IntermediateTests.cs:5]: Added to Intermediate - get list of unique numbers

**Review by stephen-atkinson — APPROVED** [review]

**Pull request merged**